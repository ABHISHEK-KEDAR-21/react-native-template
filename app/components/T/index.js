/**
 *
 * T
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { injectIntl } from 'react-intl'
import { Text } from 'react-native'

const T = ({ intl, id, values }) => (
  <Text>{intl.formatMessage({ id }, { ...values })}</Text>
)

T.propTypes = {
  id: PropTypes.string.isRequired,
  intl: PropTypes.object,
  values: PropTypes.object
}

export default compose(injectIntl)(T)
