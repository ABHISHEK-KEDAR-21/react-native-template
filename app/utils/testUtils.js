import React from 'react'
import { render } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import ConnectedLanguageProvider from 'app/containers/LanguageProvider'
import { translationMessages } from '../i18n'
import createStore from '../rootReducer'
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
})

export const renderProvider = children => {
  const { store } = createStore()
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        {children}
      </ConnectedLanguageProvider>
    </Provider>
  )
}
