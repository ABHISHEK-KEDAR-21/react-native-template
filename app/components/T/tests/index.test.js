/**
 *
 * Tests for T
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
import React from 'react'
import { renderWithIntl } from 'app/utils/testUtils'
import T from '../index'

describe('<T />', () => {
  it('Should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<T />)
    expect(baseElement).toMatchSnapshot()
  })

  it('Should contain 1 t', () => {
    const { getAllByTestId } = renderWithIntl(<T id="soemthing" />)
    expect(getAllByTestId('t').length).toBe(1)
  })
})
