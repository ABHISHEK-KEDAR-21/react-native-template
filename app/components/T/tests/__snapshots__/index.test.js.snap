// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<T /> Should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      testID="t"
    >
      some_text
    </Text>
  </View>
</View>
`;
